webapp/src/ts/services/task-notifications.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from '@angular/core';
+import * as moment from 'moment';
+import { orderBy } from 'lodash-es';
+
+import { RulesEngineService } from '@mm-services/rules-engine.service';
+import { TranslateService } from '@mm-services/translate.service';
+import { DBSyncService } from '@mm-services/db-sync.service';
+
+/* 
+** avoid overloading app with too many notifications especially at once
+** 24 for android >= 10
+*/
+const MAX_NOTIFICATIONS = 24;
+const TODAY_TIMESTAMP = 'cht-today-timestamp';
+const LATEST_NOTIFICATION_TIMESTAMP = 'cht-latest-notification-timestamp';
+
+export interface Notification {
+  _id: string,
+  authoredOn: number,
+  state: string,
+  title: string,
+  contentText: string,
+  dueDate: string,
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TasksNotificationService {
+
+  constructor(
+    private readonly rulesEngineService: RulesEngineService,
+    private readonly translateService: TranslateService,
+    private readonly dbSyncService: DBSyncService
+  ) { }
+
+  private async fetchNotifications(): Promise<Notification[]> {
+    try {
+      const today = moment().format('YYYY-MM-DD');
+      let latestNotificationTimestamp = this.getLatestNotificationTimestamp();
+      const isEnabled = await this.rulesEngineService.isEnabled();
+      const taskDocs = isEnabled ? await this.rulesEngineService.fetchTaskDocsForAllContacts() : [];
+      
+      let notifications = taskDocs
+        .filter(task => {
+          return task.state === 'Ready' && task.emission.dueDate === today && 
```

Comment:
What about tasks that are overdue? I'm curious what this system is really trying to achieve. 
If a user doesn't log in for one day, when a specific task is due, and logs in the next day when the task is overdue, they never get notified for it.